configure.ac

```diff
@@ -124,7 +124,7 @@ AS_IF([test "x$enable_threadsanitizer" = "xyes"], [
   AS_IF([test x != "x$sanitizeopts"], [
     AC_MSG_ERROR(Cannot enable multiple checkers at once)
   ])
-  sanitizeopts="thread"
+  sanitizeopts="thread -Wthread-safety"
```

Comment:
Given the severity of these warnings (and that we're instrumenting the library gradually anyway), should we upgrade this from a warning to an error for CI?

Reply:
I agree, these should be errors if possible

Reply:
that's a good idea, updated. I also separated thread safety into its own flag, since we don't necessarily want all the overhead of TSan.